src/guide-seeding.md

```diff
@@ -0,0 +1,119 @@
+# Seeding RNGs
+
+As we have seen, the output of pseudo-random number generators (PRNGs) is
+determined by their initial state.
+
+Some PRNG definitions specify how the initial state should be generated from a
+key, usually specified as a byte-sequence for cryptographic generators or,
+for small PRNGs, often just a word. We formalise this for all our generators
+with the [`SeedableRng`] trait.
+
+## The Seed type
+
+We require all seedable RNGs to define a [`Seed`] type satisfying
+`AsMut<[u8]> + Default + Sized` (usually `[u8; N]` for a fixed `N`).
+We recommend using `[u8; 12]` or larger for non-cryptographic PRNGs and
+`[u8; 32]` for cryptographic PRNGs.
+
+PRNGs may be seeded directly from such a value with [`SeedableRng::from_seed`].
+
+## Seeding from ...
+
+### Fresh entropy
+
+Using a fresh seed (direct from the OS) is easy using [`SeedableRng::from_entropy`]:
+
+```rust,editable
+use rand::prelude::*;
+use rand_chacha::ChaCha20Rng;
+
+let rng = ChaCha20Rng::from_entropy();
+```
+
+Note that this requires `rand_core` has the feature `getrandom` enabled.
+
+### Another RNG
+
+Quite obviously, another RNG may be used to fill a seed. We provide a
+convenience method for this:
+
+```rust,editable
+use rand::prelude::*;
+use rand_pcg::Pcg64;
+
+let rng = Pcg64::from_rng(thread_rng());
+```
+
+But, say you want to save a key and use it later. For that you need to be a
+little bit more explicit:
+
+```rust,editable
+use rand::prelude::*;
+use rand_chacha::ChaCha8Rng;
+
+let mut seed: <ChaCha8Rng as SeedableRng>::Seed = Default::default();
+thread_rng().fill(&mut seed);
+let rng = ChaCha8Rng::from_seed(seed);
+```
+
+**Obligatory warning**: a few simple PRNGs, notably [`XorShiftRng`],
+behave badly when seeded from the same type of generator (in this case, Xorshift
+generates a clone). For cryptographic PRNGs this is not a problem;
+for others it is recommended to seed from a different type of generator.
+[`ChaCha8Rng`] is an excellent choice for a fast, non-cryptographic
+deterministic master generator.
```

Comment:
Maybe we should mention that this is still a cryptographic generator, although with a small security margin?

Reply:
As far as I understand this is never recommended for cryptographic uses though. You think the language is confusing? I prefer not to be too verbose.

Reply:
I think it's fine with your new comment!